cedar-java-hello-world/README.md

```diff
@@ -16,7 +16,7 @@ bash config.sh
 ### Build
 
 ```shell
-git clone -b release/2.3.x https://github.com/cedar-policy/cedar-java.git
+git clone -b release/3.2.x https://github.com/cedar-policy/cedar-java.git
```

Comment:
Is the `git clone` here (and in CI) doing anything? Doesn't v3.1.2 of `cedar-java` always get pulled from Maven? Is there a way to update the build to use a local version instead?

Reply:
> Is the git clone here (and in CI) doing anything? Doesn't v3.1.2 of cedar-java always get pulled from Maven?

My impression is that we pull `cedar-java` here locally for the CedarJavaFFI dynamic lib built, not the CedarJava library.

While the `SampleJavaClass` source points to the v3.1.2 Maven CedarJava to resolve the Cedar objects, that library will look for `$CEDAR_JAVA_FFI_LIB` locally when [LibraryLoader](https://github.com/cedar-policy/cedar-java/blob/main/CedarJava/src/main/java/com/cedarpolicy/loader/LibraryLoader.java#L38) is used. To populate this env var, we need to run the [`config.sh`](https://github.com/cedar-policy/cedar-examples/blob/release/3.2.x/cedar-java-hello-world/config.sh#L6-L10) script in `cedar-java-hello-world` to point to the local CedarJavaFFI dynamic library we built.

I selected the `3.2.x` branch here as we need the [CedarJavaFFI cedar lang version](https://github.com/cedar-policy/cedar-java/blob/release/3.2.x/CedarJavaFFI/src/interface.rs#L107) to match [that of the CedarJava library from Maven](https://github.com/cedar-policy/cedar-java/blob/release/3.1.x/CedarJava/src/main/java/com/cedarpolicy/AuthorizationEngine.java#L84). Also this corresponds with the `cedar-examples` `release/3.2.x` branch.

> Is there a way to update the build to use a local version instead?

If we wanted, we could alter the `cedar-java-hello-world` `build.gradle` to point to the local CedarJava build:
```
dependencies {
...
//implementation 'com.cedarpolicy:cedar-java:3.1.2'
implementation files('cedar-java/CedarJava/build/libs/CedarJava.jar')
...
}
```
This could either be a change in `config.sh` or a separate bash script.

Reply:
Thanks for clarifying! I thought that the `CedarJavaFFI` library was distributed on Maven so we didn't need to build it locally anymore? Or is that not the case? It seems like with the current setup you could end up with a `CedarJavaFFI` version that is out of sync with the package on Maven.

I think what we should do for this example is similar for what we do in the [wasm example](https://github.com/cedar-policy/cedar-examples/blob/main/README.md):
* In the README, give the simplest instructions first (ideally, not requiring a local build)
* Then provide instructions for how to use a local build
* In CI, update the gradle file to use a local build ([example](https://github.com/cedar-policy/cedar-examples/blob/53a9e789c5d7d26b595bb0c87f8d1379df60ec77/.github/workflows/build_wasm_example_reusable.yml#L42))

Reply:
> I thought that the CedarJavaFFI library was distributed on Maven so we didn't need to build it locally anymore? Or is that not the case?

I believe that is not the case. Confirmed with Aaron and pending a check with Andrew. Without that local CedarJavaFFI environment, I'll always get a null pointer exception from the library loader [here](https://github.com/cedar-policy/cedar-java/blob/release/3.1.x/CedarJava/src/main/java/com/cedarpolicy/loader/LibraryLoader.java#L33-L38) and even the [cedar-java-ffi](https://github.com/cedar-policy/cedar-java/blob/release/3.1.x/CedarJava/src/main/java/com/cedarpolicy/loader/LibraryLoader.java#L27C49-L27C63) lib cannot be resolved when [CEDAR_JAVA_FFI_LIB](https://github.com/cedar-policy/cedar-java/blob/release/3.1.x/CedarJava/src/main/java/com/cedarpolicy/loader/LibraryLoader.java#L25) is not set.

> It seems like with the current setup you could end up with a CedarJavaFFI version that is out of sync with the package on Maven.

Agreed, and perhaps there's a way to package the CedarJavaFFI in the Maven release without requiring the local build.

> I think what we should do for this example is similar for what we do in the [wasm example](https://github.com/cedar-policy/cedar-examples/blob/main/README.md)

Agreed with you on these, but yeah for item 1 as it stands we can't without pulling `cedar-java` locally. I'd say we can have two set of instructions for just building CedarJavaFFI and then building local CedarJava as well.